node/src/gql/mutation/login-google.ts

```diff
@@ -93,8 +93,24 @@ export const loginGoogle = {
         },
         {
           upsert: true,
+          new: true,
         }
       );
+      const mentorId = mentor._id;
+      // Append new mentorIds to users' mentorId list
+      if (user.mentorIds.indexOf(mentorId) === -1) {
```

Comment:
Instead of doing it this way, I think we can just check if the User has any mentorIds already.
If there is a mentorId, we don't have to do anything. If there isn't (the user's first time logging in), we just have to create a new mentor and add it to the user.